[styles] Rework parking icons

RedAuburn opened the pull request:
- Remove background from all parking icons
- Render parking=lane & street_side smaller (fixes https://github.com/organicmaps/organicmaps/issues/3863)

![sfdsfsf](https://user-images.githubusercontent.com/26939824/202042417-42237a43-233b-4941-9b22-963776a943d2.png)

see https://github.com/organicmaps/organicmaps/pull/3896/commits/f53327315c1dc08cec64e8ef4f42e0445e20e381 for icons before and after
(requires map regen for smaller street_side/lane parking)

biodranik commented:
Thanks! Looks better. What's in the bottom right?

endim8 commented:
> Thanks! Looks better. What's in the bottom right?

The bike parking:
<img width="1505" alt="Screenshot 2022-11-16 at 10 51 44" src="https://user-images.githubusercontent.com/26939824/202161714-12edab36-6399-4043-a709-b22ee5ab719e.png">
(motorcycle parking looks the same)

biodranik commented on data/types.txt:
```diff
@@ -355,7 +355,7 @@ mapswithme
 amenity|parking|underground
 *amenity|parking|underground
-mapswithme
+*amenity|parking|street_side
```
@vng please review and merge if it's ok.

vng commented:
Are we sure that car parking without background is better? Agree that bicycles look better now, but car parking is something _important_ and should be visible one of the top types, especially in driving mode ..

vng commented:
And white P letter on blue background is a unique parking sign, no?

endim8 commented:
Original | Only replace street_side | New

![types](https://user-images.githubusercontent.com/26939824/202228527-3a4e08aa-01a7-4907-854d-95f4507674ed.png)

endim8 commented:
> E.g. use a blue border for street-side parking, without any icon? 

i'm not quite sure what you mean, just a blue rectangle?

biodranik commented:
What if we remove icons completely for street-side parking?

matheusgomesms commented:
> What if we remove icons completely for street-side parking?

This is an option that some members that I talked to wanted to see on Carto, meaning that would look just like [`amenity=parking_space`](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dparking_space). Another option is to remove the yellowish background, with a smaller P.

dvdmrtnz commented:
> Original | Only replace street_side | New
> 
> ![types](https://user-images.githubusercontent.com/26939824/202228527-3a4e08aa-01a7-4907-854d-95f4507674ed.png)

I think the "Only replace street_side" is the best option. Let's leave the normal parkings as they are and the street-side parkings will get the smaller P without background. 

Also I think I would start drawing the street-side parkings icon at zoom level 18 instead of 17.

tjur0 commented:
I also think the "Only replace street_side" is the best option. I do think we should leave the bicycle parking alone for now as there is no tagging to distinguish between street side and bigger parking lots for bikes.

mpawelski commented:
Great! thanks @endim8 for working on it.

My two cents:

> Are we sure that car parking without background is better? Agree that bicycles look better now, but car parking is something _important_ and should be visible one of the top types, especially in driving mode ..
> 
> And white P letter on blue background is a unique parking sign.

P letter on blue background is definitely a recognizable sign but I think users won't have any problems with guessing what a blue P letter means anyway.
But I might be biased. I've looked on such parking icons on Carto OSM for years and might just get used to it. Maybe showing OM to completely new user and asking what does this "P" icon mean could be hard for this user to answer. But I really doubt it. Most user should "get it".

One of the reason I like that we consider removing this background icons is that I personally don't care that much about parking icons when _I'm_ using OM (I use it for on-foot navigation mostly). But I still want to see and recognize them because _sometimes_ I care.

And I agree with the point that this icons should be clearly visible in the driving mode. When you drive you should probably recognize this icons with just a glance. So I'm voting for keeping the old icons in the driving mode.



> Original | Only replace street_side | New
> 
> ![types](https://user-images.githubusercontent.com/26939824/202228527-3a4e08aa-01a7-4907-854d-95f4507674ed.png)

I don't like the "Only replace street_side". I don't think the _purpose_ of normal parking and street_side parking is that different to justify so different icons.

I prefer the "New" version. I would even make street_side icon slightly bigger (but not too much), because bike parking looks slightly too big when displayed next to street_side parking. This is something I [dislike](https://github.com/organicmaps/organicmaps/issues/3863#issuecomment-1315209729) in OSM Carto now.
But this is just my nitpicking about styles. I still like the icons and their sizes in "New" much more than what is in "Original".

> What if we remove icons completely for street-side parking?

I don't think the current color for parking is conveying the meaning that well. Without the icon I wouldn't guess that this area is a parking just by looking at this yellow color.

Misalf-git commented:
> > Original | Only replace street_side | New
> > 
> > ![types](https://user-images.githubusercontent.com/26939824/202228527-3a4e08aa-01a7-4907-854d-95f4507674ed.png)
> 
> I think the "Only replace street_side" is the best option. Let's leave the normal parkings as they are and the street-side parkings will get the smaller P without background. 
> 
> Also I think I would start drawing the street-side parkings icon at zoom level 18 instead of 17.

I second that.
My two cents: Areas with many parking spaces deserve the easily spottable bigger icon (as it is currently) where it's more likely to find a free spot to park your car.
Street side parking spaces usually hold space for just a few vehicles, or even only one, but is't good to know if there are any.

mpawelski commented:
> > > Original | Only replace street_side | New
> > > ![types](https://user-images.githubusercontent.com/26939824/202228527-3a4e08aa-01a7-4907-854d-95f4507674ed.png)
> > 
> > 
> > I think the "Only replace street_side" is the best option. Let's leave the normal parkings as they are and the street-side parkings will get the smaller P without background.
> > Also I think I would start drawing the street-side parkings icon at zoom level 18 instead of 17.
> 
> I second that. My two cents: Areas with many parking spaces deserve the easily spottable bigger icon (as it is currently) where it's more likely to find a free spot to park your car. Street side parking spaces usually hold space for just a few vehicles, or even only one, but is't good to know if there are any.
> 
> My two cents: Areas with many parking spaces deserve the easily spottable bigger icon (as it is currently) where it's more likely to find a free spot to park your car.

I agree that normal parking deserves more "spottable" icon, but why it should be a _different_ icons than the one used for `street_side/lane` parking? 
The "New" options is giving you bigger, more spottable icon for normal parking, while keeping the `street_side/lane` parking places still visible, but less distracting.

For me having two icons, one with background and one without, suggest that these parking places differ in something more than just a size (which in practice is the main practical difference between `street_size/lane` and other types of [Key:parking](https://wiki.openstreetmap.org/wiki/Key:parking)).

When I would see on map this two different parking icons I would wonder what's the difference between them. Maybe one is a public parking and the other is not, one is paid and the other is free, or something else. 

But `street_side/lane` is just an usual parking, it just happens that they are often much smaller and there are areas where there is many of them mapped (micromapping FTW!) . Rendering this parking areas with smaller icons just makes the map a bit more cleaner to look at (which IMO is the goal of this PR) without suggesting that this parking places have some bigger differences.

Misalf-git commented:
> When I would see on map this two different parking icons I would wonder what's the difference between them. Maybe one is a public parking and the other is not, one is paid and the other is free, or something else. 

Keeping graphical consistancy sounds logical. I'd certainly welcome smaller icons for street side parking, though, and when zooming-in increase their size at a later level, if at all.

endim8 commented:
what is the consensus on this? not sure what to add/remove

Misalf-git commented:
I think Carto does it well.
Basically "street_side" smaller and later.
Not too sure about the non-background variant. May fit to "street_side" only.

biodranik commented:
So the main question is "will users recognize that different parking icons with and without backgrounds in different modes mean the same?". Or "will users recognize that icons without background mean parking in all modes?".
On the opposite side there is this question: "is the current consistent icon with a background really inconvenient for users?"

endim8 commented:
I've changed back to the original icon for standard parking, while rendering the new icon for `lane` & `street_side` bigger and on a higher zoom. 
IMO this is the most balanced feeling, as when searching for parking in an unknown area, users are more likely to look for a car park. Alternatively, when a user is routing to a specific place eg. a friend's house, they'll want to use street_side parking nearby their destination.

https://user-images.githubusercontent.com/26939824/210673101-99f227e2-a686-4a03-9c4d-971ea7204972.mp4

Sidenote: 
Working on this really highlights how badly OM needs a function for handling access tags (to the point of preventing better drawing styles), as each access tag needs to be considered for each parking type.  For example, 6 parking types and 4 access tags results in having to add 24 parking types (48 drawing rules for node & area). It gets messy very quickly.

endim8 commented:
the parking on the right is mistagged, but is useful to show how it used to look
![signal-2023-01-05-001128_004](https://user-images.githubusercontent.com/26939824/210673564-444920f8-f92e-4a3d-9bc0-33626d3971c6.png)

biodranik commented:
Thanks! Do we need that P letter at all for street-side parking? Isn't it obvious even without the icon?

endim8 commented:
> Thanks! Do we need that P letter at all for street-side parking? Isn't it obvious even without the icon?

i thimk it's a good idea to keep it, but just at a very high zoom level :) in my experience, I didn't immediately associate yellow with parking either. does anyhing need fixing? or is thins good to merbger?

endim8 commented:
btw thank you for organicmaps youre doing a greatjob

endim8 commented:
@biodranik @vng PTAL, anything need changing?

biodranik commented:
The initial issue was about the cluttered map. Will it help to reduce the clutter if street-side parking icons will be displayed from Z17 or Z18? (And a smaller size would be helpful, too).

muralito commented:
> also, while it's true streetside parking and car parks serve the same basic purpose
IMO your assumption is not valid worlwide. Let me explain, in the countries/cities where i usually drive, you can legally park streetside, but in some suburbs or specific hours you search for a car park for security reasons because you pay not only for the space but for the supervision service, because if you left your car in the street, when you came back, maybe it hasn't wheels, or battery, or the car is not at all.
So, i think the basic purpose in those cases is not parking a car, but the security you get.

And as such the "supervised" tag has an important meaning in searches for parkings.

biodranik commented:
@pastk can you please take a look here?

pastk commented:
1. icons consistency is important, i.e. use the same style for all parking types and in all modes (map browsing / navigation)
2. smaller icons for lesser parkings (e.g. street side) shown on higher zooms with a smaller displacing priority - good!

Either the current squared P or the new P without background are fine, its mostly a matter of getting used to it IMO.
The latter one is cleaner and will allow for easier modification like adding a small `$` in the corner for paid parkings or adding a `^` on top for covered / underground parkings.
Also bicycle/motorcycle parkings w/o background look much cleaner/better!

biodranik commented:
Can we make a fresh build and try it?

RedAuburn commented:
Rebased and now only using the P without background 👍

generated MWM for testing: https://drive.proton.me/urls/RKB1JP06KC#2H8ok1MoEZft

biodranik commented on data/styles/clear/style-night/symbols/parking_entrance-m.svg:
Are these icons transparent?

RedAuburn replied:
> 

they are, but it makes more sense for the P to be opaque 👍 this is an issue with lots of night icons though, some are opaque and some are transparent

biodranik commented on data/styles/clear/style-night/symbols/bike-parking-m.svg:
Can the bicycle shape be simplified, to be easier recognizable and distinguishable from motorbike?
This is a sample from maki icons:
![bicycle](https://github.com/organicmaps/organicmaps/assets/170263/68674b14-b7ad-4abd-a231-e58c6b9a5df4)

RedAuburn replied:
changed to the maki icon 👍

biodranik commented on data/styles/clear/style-clear/symbols/parking_pay-m.svg:
Is it an international paid symbol? It is not familiar to me. Maybe use a dollar sign instead?

pastk replied:
Ahh, the small circles are supposed to be coins - it took me awhile to get it :) They're indistinguishable at this size.
The international sign is https://thumbs.dreamstime.com/b/paid-parking-road-sign-light-background-65685403.jpg

But it'll be very hard to fit even two "coins" into the icon.

biodranik commented on data/styles/clear/style-clear/symbols/motorcycle-parking-m.svg:
Can a motorbike have more different shape than a bike, for easier distinguishing it on the screen, especially for smaller icons?

biodranik reviewed (commented):
Thanks! It would be great to test it on a mobile screen too...

pastk reviewed (changes requested):
It'd be good to amend the vehicle style in this PR also.

Please add the new -street_side and -lane mappings into types_strings.txt and categories.txt.

pastk commented:
> 
> generated MWM for testing: https://drive.proton.me/urls/RKB1JP06KC#2H8ok1MoEZft

I've tested it and it looks good overall, thanks!

ATM `parking-s.svg` is not used at all. Maybe use it for -lane and -street_side for z17-? And lower its priority.
(And for regular parkings keep keep the -m icon z16-).

Also ATM private parkings and [amenity=parking][access=no] use the same style as regular ones. I guess we can show a small icon at z19- only or some special icon or maybe no icon at all..

pastk commented:
> Please add the new -street_side and -lane mappings into types_strings.txt and categories.txt.

@vng actually in categories.txt we have
```amenity-parking|@parking```
and searching for "Parking" results in e.g. amenity-parking-private and amenity-parking-lane (so sub-types are matched), though not all of them get a blue marker somehow (while the icons are visible and its a plenty of space around on z19).

RedAuburn commented:
thank you for the reviews 😊 will fix the issues

pastk commented:
> @vng actually in categories.txt we have `amenity-parking|@parking` and searching for "Parking" results in e.g. amenity-parking-private and amenity-parking-lane (so sub-types are matched), though not all of them get a blue marker somehow (while the icons are visible and its a plenty of space around on z19).

So apparently there is no need to add to categories.txt, only into types_strings.txt.

RedAuburn commented on data/mapcss-mapping.csv:
```diff
@@ -428,7 +428,7 @@ highway|unclassified|area;[highway=unclassified][area?];;name;int_name;354;
 deprecated:amenity|parking|underground|fee:01.2020;[amenity=parking][location=underground][fee?],[amenity=parking][parking=underground][fee?];x;name;int_name;356;amenity|parking|underground
 amenity|parking|underground;[amenity=parking][location=underground],[amenity=parking][parking=underground];;name;int_name;357;
-deprecated|deprecated;358;x
+amenity|parking|street_side;[amenity=parking][parking=street_side],[amenity=parking][parking=street_side];;name;int_name;358;
```
fixed

RedAuburn commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,27 +3010,46 @@ area|z15-[amenity=sanitary_dump_station]
-{icon-image: zero-icon.svg;}
+node|z16-[amenity=parking],
+area|z16-[amenity=parking]
+{icon-image: parking-m.svg;icon-min-distance: 25;}
```
fixed

RedAuburn commented:
> Please add the new -street_side and -lane mappings into types_strings.txt

i'll do this once the styles are done, as otherwise the comments will be filled with translation reviews 😅

I've added styles for `[amenity=parking][access=no]`, but it looks like it's ignored in favour of this mapcss case:
 `[!oneway] matches [oneway=no] only or absence of this tag.`

But it should only match the explicit tagging of `no`, not the tag absence.

RedAuburn commented:
updated drules, changed bicycle-parking icon, added private parking icon, improved parking entrance icon, improved paid parking icon

biodranik commented on data/styles/clear/style-clear/symbols/parking_pay-m.svg:
Would $ be more readable (especially on lower/smaller resolutions) if it is blue, not white?

Did you also check how all icons look in dark mode?

RedAuburn replied:
you're right, changed the icon to have the $ in blue and it looks better.

all icons look good in dark mode :)

biodranik commented on data/styles/clear/style-clear/symbols/parking_private-m.svg:
Maybe P of a red color will be better for private parking?

RedAuburn replied:
![image](https://github.com/organicmaps/organicmaps/assets/26939824/b52fed1d-efe0-4e6a-83d8-c749b6a9e179)
this looks good 👍

biodranik commented on data/types.txt:
```diff
@@ -353,9 +353,9 @@ mapswithme
 *highway|unclassified|area
 *natural|volcano
-amenity|parking|underground
+*amenity|parking|fee:01.2020
```
What is this type?

RedAuburn replied:
oops, was a depreciated type. fixed 👍

biodranik reviewed (commented):
I didn't test it yet, can someone share a few screenshots of how it looks in dense cities and in the dark mode?

RedAuburn commented:
> I didn't test it yet, can someone share a few screenshots of how it looks in dense cities and in the dark mode?

here are a few quick ones, but it's late here now so i'll do more detail tomorrow :)

(rendered in the desktop application, and scaling is still not right (https://github.com/organicmaps/organicmaps/issues/4836) so it's not exactly how it'll look in mobile)
![Screenshot from 2023-08-07 00-34-34](https://github.com/organicmaps/organicmaps/assets/26939824/7beff15e-106d-43d1-81e0-bff2525f36f0)
![Screenshot from 2023-08-07 00-33-42](https://github.com/organicmaps/organicmaps/assets/26939824/398e7a87-eeea-4dc4-acd5-574d1c06aea4)
![Screenshot from 2023-08-07 00-33-10](https://github.com/organicmaps/organicmaps/assets/26939824/9a5fe00d-b011-4161-bca1-9dd11679ffbb)
![Screenshot from 2023-08-07 00-30-25](https://github.com/organicmaps/organicmaps/assets/26939824/97d2e4ba-67ac-4ccd-868e-0b26fc2de515)
![Screenshot from 2023-08-07 00-26-01](https://github.com/organicmaps/organicmaps/assets/26939824/a1a9ab5a-abf4-4dac-b18c-085a7f869640)
![Screenshot from 2023-08-07 00-25-24](https://github.com/organicmaps/organicmaps/assets/26939824/275e124f-bcbc-4a3f-8855-6e30e7104437)
![Screenshot from 2023-08-07 00-25-10](https://github.com/organicmaps/organicmaps/assets/26939824/2df5de49-1cc3-46b5-8420-0bc34e894f0b)

biodranik commented:
Please fix failing tests:
```
 Running osm_type_test.cpp::TestWithClassificator_OsmType_MergeTags
FAILED
generator_tests/osm_type_test.cpp:1184 TEST(params.IsTypeExist(GetType({"amenity", "parking", "underground"}))) ReversedGeometry: false; Metadata []; AddressData [Street = ""; Postcode = ""]; Types: amenity-parking; 
Test took 65 ms
```

```
Running osm_type_test.cpp::TestWithClassificator_OsmType_ComplexTypesSmoke
FAILED
generator_tests/osm_type_test.cpp:2631 TEST(params.IsTypeExist(GetType(type.first))) ([3: amenity parking underground ], [2: amenity=parking location=underground ]) ReversedGeometry: false; Metadata []; AddressData [Street = ""; Postcode = ""]; Types: amenity-parking; 
Test took 346 ms
```

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,50 +3010,64 @@ area|z15-[amenity=sanitary_dump_station]
+node|z16-[amenity=parking][parking=underground],
+area|z16-[amenity=parking][parking=underground],
+node|z16-[amenity=parking][location=underground],
+area|z16-[amenity=parking][location=underground]
```
'[location=underground]' line is enough;
(in fact, kothic matches only the first option from the mapping line `amenity|parking|underground;[amenity=parking][location=underground],[amenity=parking][parking=underground];;name;int_name;357;`)

pastk replied:
Also it makes underground parkings disappear completely?
What was your plan? To show them on later zooms than surface parkings?

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,50 +3010,64 @@ area|z15-[amenity=sanitary_dump_station]
+{icon-image: parking_pay-m.svg;icon-min-distance: 5;}
+node|z18-[amenity=parking][fee?],
+area|z18-[amenity=parking][fee?]
+{icon-image: parking_pay-m.svg;font-size: 11;}
```
icon name is inherited from the z16- line

pastk commented on data/styles/clear/style-clear/symbols/bicycle-parking-m.svg:
hmm, somehow in github it shows for me not square, but a long vertical rectangle...
I wonder if it'll create issues

RedAuburn replied:
not sure why github does that for some icons, but there's no reason for it in the SVG code & the canvas is the correct size in inkscape: 
![image](https://github.com/organicmaps/organicmaps/assets/26939824/3160df19-e4dc-48d4-a265-7eb0a04e2177)

pastk commented on data/styles/clear/style-clear/symbols/parking_private-m.svg:
this one shows as a vertical rectangle too

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1053,33 +1053,35 @@ area|z16-[amenity=water_point],
+{icon-image: ; icon-min-distance: 3;z-index: 20000;}
+node|z17-[amenity=parking][access=permissive],
+node|z17-[amenity=parking][access=private]
+{icon-image: ;z-index: 6;icon-min-distance: 7;}
```
no point to set z-index or icon-min-distance for an empty icon;

for z18- please use a private parking icon

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1053,33 +1053,35 @@ area|z16-[amenity=water_point],
+{icon-image: ;}
+node|z18-[amenity=parking_entrance],
+area|z18-[amenity=parking_entrance]
+{icon-image: parking_entrance-m.svg;font-size: 11;}
```
vehicle style uses a different (bigger) range of "standard" font sizes than main;
maybe check e.g. amenity=fuel and do similarly

pastk reviewed (changes requested):
Good progress!

Please add paid parkings and street_side/lane parkings to the vehicle style too!

I think motorcycle parkings should be added there too.

pastk commented:
> > Please add the new -street_side and -lane mappings into types_strings.txt
> 
> i'll do this once the styles are done, as otherwise the comments will be filled with translation reviews sweat_smile

For now you can just add stubs with `ref = type.amenity.parking` there. And make a separate PR for detailed translations later.

> I've added styles for `[amenity=parking][access=no]`, but it looks like it's ignored in favour of this mapcss case: `[!oneway] matches [oneway=no] only or absence of this tag.`
> 
> But it should only match the explicit tagging of `no`, not the tag absence.

Right, this is an old kothic matching bug.
So I guess comment access=no styles for now... (its 0.3% of cases so not too bad)

pastk commented:
> here are a few quick ones, but it's late here now so i'll do more detail tomorrow :)

Red P's of private parkings look more prominent/bright than regular blue P's. Should be vice versa :)

Maybe make private parkings icons
- small always
- less bright/saturated

RedAuburn commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,50 +3010,64 @@ area|z15-[amenity=sanitary_dump_station]
+area|z16-[amenity=parking][parking=street_side],
+node|z16-[amenity=parking][parking=lane],
+area|z16-[amenity=parking][parking=lane]
+{icon-image: ;icon-min-distance: 1;}
```
Will not setting a min-distance force it to 0? I thought values have to be overwritten otherwise it'll inherit the min-distance from the base `[amenity=parking]`? Similar to `icon-image: ;`

pastk replied:
Yeap, you're right!

Misalf-git commented:
I'd prefer transparecy like OSM Carto. Maybe small also.

RedAuburn commented:
argh the tests are still failing, i'll try and fix.

pastk commented on data/mapcss-mapping.csv:
```diff
@@ -427,8 +427,8 @@ highway|cycleway|permissive;[highway=cycleway][access=permissive];;name;int_name
 highway|unclassified|area;[highway=unclassified][area?];;name;int_name;354;
 natural|volcano;355;
 deprecated:amenity|parking|underground|fee:01.2020;[amenity=parking][location=underground][fee?],[amenity=parking][parking=underground][fee?];x;name;int_name;356;amenity|parking|underground
-amenity|parking|underground;[amenity=parking][location=underground],[amenity=parking][parking=underground];;name;int_name;357;
```
hmm, why did you remove the `[amenity=parking][location=underground]` mapping?

RedAuburn replied:
oohhh thank you! this is why the tests are failing 👍

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,50 +3010,65 @@ area|z15-[amenity=sanitary_dump_station]
+/* area|z16-[amenity=parking][access=no], */
+node|z17-[amenity=parking_entrance][access=permissive],
+node|z17-[amenity=parking_entrance][access=private]
+{icon-image: ;}
```
`[amenity=parking_entrance][access=permissive]` is defined later in the file, so there is nothing to override at this point;

in fact the later declaration will override this one and those entrances will have an icon

RedAuburn replied:
good catch, fixed 👍

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -2932,14 +2932,14 @@ node|z15-[amenity=sanitary_dump_station],
 node|z18-[amenity=parking],
 area|z18-[amenity=parking],
+node|z18-[amenity=bicycle_parking],
+area|z18-[amenity=bicycle_parking],
```
did we want to demote bicycle parkings from z17 to z18? I don't remember it in the discussion - cyclists won't be happy :)

RedAuburn replied:
it wasn't mentioned in the discussion, but in cities they can cause a lot of visual clutter. i'll leave it at z17 for now 👍

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1052,34 +1052,53 @@ area|z16-[amenity=water_point],
+
+node|z15-[amenity=parking][fee?],
+area|z15-[amenity=parking][fee?]
+{icon-image: parking_pay-m.svg;icon-min-distance: 5;font-size: 12.5;}
```
its enough to override the icon name only, everything else will come from declarations above

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1052,34 +1052,53 @@ area|z16-[amenity=water_point],
+node|z17-[amenity=parking][parking=street_side],
+area|z17-[amenity=parking][parking=street_side],
+node|z17-[amenity=parking][parking=lane],
+area|z17-[amenity=parking][parking=lane]
```
nit-picking: its better to end with a comma always (here and in other changed lines of this PR) - this way there will be less "diff noise" next time when adding another type into the list

RedAuburn replied:
ah ok 👍 fixed and will do in future

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1052,34 +1052,53 @@ area|z16-[amenity=water_point],
+{icon-image: parking-m.svg;font-size: 12.5;}
+
+node|z17[amenity=parking_entrance],
+area|z17[amenity=parking_entrance]
```
let's show it from z16 in the vehicle style

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1052,34 +1052,53 @@ area|z16-[amenity=water_point],
+{icon-image: parking_entrance-m.svg;font-size: 14.5;}
+node|z17-[amenity=parking][access=private],
+area|z17-[amenity=parking][access=private]
+{icon-image: parking_private-m.svg;font-size: 14.5;}
```
merge it with a block just above?

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -1053,33 +1053,48 @@ area|z16-[amenity=water_point],
+area|z15-[amenity=parking][parking=street_side],
+node|z15-[amenity=parking][parking=lane],
+area|z15-[amenity=parking][parking=lane],
+{icon-image: ;}
```
now a lone caption gets displayed at z16, please remove it for z16 via `text: none;`

also we have inconsistency:
in the main style [access=permissive] has the same style as regular parkings, but here all its icons are removed
I think makes sense to do as in the main

RedAuburn replied:
fixed and fixed 👍

pastk reviewed (changes requested):
Looks good on the screen overall!

Except the new bicycle icon is much bigger than a motorcycle and almost as prominent as a regular P - bicycles should be smaller IMO.

Hint: after chaging *.mapcss files and regenerating take a look into changes made to drules*.txt files, many inconsistencies will be obvious.

RedAuburn commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,51 +3010,56 @@ area|z15-[amenity=sanitary_dump_station]
-
+node|z16-[amenity=parking],
+area|z16-[amenity=parking],
+{icon-image: parking-m.svg;icon-min-distance: 5;}
```
-s returned to z16 👍

pastk commented:
Please rebase onto a new master and add the new types into `[clear|vehicle]/includes/priorities_4_overlays.prio.txt` and make them lower than regular parkings there.

pastk commented:
@RedAuburn we want to include your PR into the upcoming release, will you be able to finish it in a couple of days? Or I can help finish it?

RedAuburn commented:
> @RedAuburn we want to include your PR into the upcoming release, will you be able to finish it in a couple of days? Or I can help finish it?

Hey sorry for the delay, i'm up in Scotland at the moment and there's almost no internet 😅 i'll do the last fixes now 👍

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -3010,51 +3010,56 @@ area|z15-[amenity=sanitary_dump_station]
-{icon-image: zero-icon.svg;}
+node|z17-[amenity=parking_entrance][access=private],
+area|z17-[amenity=parking_entrance][access=private],
+{icon-image: ;}
```
I checked the code and it should work as intended:
if there will be no styles defined for amenity-parking_entrance-private then those features will be not included into mwms
(i.e. they won't re-appear as an amenity-parking_entrance)

pastk replied:
I checked the code and it should work as intended:
if there will be no styles defined for amenity-parking_entrance-private then those features will be not included into mwms
(i.e. they won't re-appear as an amenity-parking_entrance)

RedAuburn commented:
Resolved review comments, made bicycle icon same height as motorcycle, added an -s version of parking_pay, added types to the new priorities.txt

@pastk feel free to fix anything if it's wrong :)

RedAuburn commented:
> could you please (only if you have time) check how it is now in a road full of street_side parking? Such as 40.209607, -8.399199?

Here's a before/after of a similar area :)

![bitmap](https://github.com/organicmaps/organicmaps/assets/26939824/18072d4b-0462-4a70-9036-d4df0e5f5da4)

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -2492,50 +2492,59 @@ area|z15-[amenity=sanitary_dump_station]
-{icon-image: zero-icon.svg;}
+node|z17-[amenity=parking_entrance][access=private],
+area|z17-[amenity=parking_entrance][access=private],
+{icon-image: ;icon-min-distance: 0;}
```
no need to reset icon-min-distance: as it was not declared for [amenity=parking_entrance] above

and you didn't add `text: none;` to disable the captions :)

RedAuburn replied:
oops, fixed 👍

pastk commented on data/styles/clear/include/Icons.mapcss:
```diff
@@ -2492,50 +2492,59 @@ area|z15-[amenity=sanitary_dump_station]
-node|z17[amenity=parking_entrance],
-area|z17[amenity=parking_entrance]
+area|z18-[amenity=parking],
+{icon-image: parking-m.svg;font-size: 11;icon-min-distance: 10}
```
for regular and paid parkings please use the -m icon from z17-

otherwise at z17 both regular and street side parkings will looks the same with the -s icon

pastk replied:
even bicycle parking uses -m at z17 :)

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -886,27 +886,45 @@ area|z16-[amenity=water_point],
-{icon-image: parking-m.svg; icon-min-distance: 3;}
-
+area|z15-[amenity=parking],
+{icon-image: parking-m.svg;icon-min-distance: 10;z-index: 20000;font-size: 12.5;}
```
please remove z-index, no more z-indexes in mapcss files! :)
all priorities were moved into prio.txt files

biodranik replied:
@pastk can kothic fail in such cases or warn at least?

pastk replied:
it actually just ignores `z-index:` now, so my nit was just about keeping the file clean

pastk commented on data/styles/vehicle/include/Icons.mapcss:
```diff
@@ -886,27 +886,45 @@ area|z16-[amenity=water_point],
+
+node|z16-[amenity=parking_entrance],
+area|z16-[amenity=parking_entrance],
+{icon-image: parking_entrance-m.svg; font-size: 12.5;}
```
amenity-parking_entrance should be added into `vehicle/include/priorities_4_overlays.prio.txt`

pastk commented:
@RedAuburn if you don't have time please let me know and I'll finish the work (I'll have to open a new PR though I guess)

RedAuburn commented:
Resolved review, thank you for your reviews & being so patient @pastk 😊

pastk reviewed (approved):
I tested it and looks good on the screen!
Thank you for the great PR!
